### PR TITLE
refactor: Remove the redundant RelayerProcessor in monitor bot code

### PR DESCRIPTION
**Pull request by kevinuma**

### CODE REVIEW

File: src/monitor/Monitor.ts [hunk 1]
```diff
@@ -78,21 +75,20 @@ export class Monitor {
...
+    const chainIds = Object.keys(this.clients.spokePoolClients);
     this.logger.debug({ at: "AcrossMonitor#UnknownRelayers", message: "Checking for unknown relayers", chainIds });
     for (const chainId of chainIds) {
-      const relayEvents: any = await this.relayerProcessor.getRelayedEventsInfo(
-        this.clients.spokePools[chainId],
+      const fills = await this.clients.spokePoolClients[chainId].getFillsWithBlockInRange(
         this.spokePoolsBlocks[chainId].startingBlock,
         this.spokePoolsBlocks[chainId].endingBlock
       );
-      for (const event of relayEvents) {
```
Comment (chrismaree) [inline src/monitor/Monitor.ts:89]: this is the exact kind of refactor I wanted to see.

File: src/clients/MonitorClientHelper.ts [hunk 2]
```diff
@@ -1,26 +0,0 @@
-import { SpokePool } from "@across-protocol/contracts-v2";
```
Comment (chrismaree) [inline src/clients/MonitorClientHelper.ts:1]: good job on moving this. this is more consistant with the relayer and data worker

File: test/Monitor.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      spokePoolClients: [spokePoolClient_1, spokePoolClient_2],
+    });
+  });
+
+  it("Monitor should log unknown relayers", async function () {
+    await updateAllClients();
+    await monitorInstance.update();
+    await monitorInstance.checkUnknownRelayers();
+    const unknownRelayerMessage = "Unknown relayer 😱";
+    expect(lastSpyLogIncludes(spy, unknownRelayerMessage)).to.be.false;
```
Comment (chrismaree) [inline test/Monitor.ts:65]: thanks for adding the tests! I know using this `lastSpyLogIncludes` is kinda painful.

File: test/Monitor.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+describe("Monitor", async function () {
+  beforeEach(async function () {
+    ({
+      configStoreClient,
+      hubPool,
+      dataworker,
+      depositor,
+      erc20_1,
+      erc20_2,
```
Comment (chrismaree) [inline test/Monitor.ts:38]: nit: do you need all these variables? `erc20_2` looks like it's not used. you might consider decreasing these and you might be able to name them more cleanly as well.

File: test/Monitor.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    // Send a deposit and a fill so that dataworker builds simple roots.
+    const deposit = await buildDeposit(
+      configStoreClient,
+      hubPoolClient,
+      spokePool_1,
+      erc20_1,
+      l1Token_1,
+      depositor,
+      constants.destinationChainId,
+      constants.amountToDeposit
```
Comment (chrismaree) [inline test/Monitor.ts:76]: nit: if you import `destinationChainId` and `amountToDeposit` directly from the `"./constants"` export then you might be able to get this 9 line call to `buildDeposit` down to one line. I like having these kinds of things be one line, if posible.

**Review by chrismaree — APPROVED** [review]
looks good! I think you can make your code a bit more concise but overall I'm happy.

**Comment by chrismaree** [comment]
one quick question before you merge: did you try running this locally for a bit to check it picks up the prod relayer?

File: test/Monitor.ts (hunk 4 above)
Comment (kevinuma) [reply test/Monitor.ts]: Unfortunately yes, except for erc20_2, which I removed.

File: test/Monitor.ts (hunk 5 above)
Comment (kevinuma) [reply test/Monitor.ts]: Updated

**Comment by kevinuma** [comment]
> one quick question before you merge: did you try running this locally for a bit to check it picks up the prod relayer?

Yes I re-ran a few times after changes and didn't see any error log when relays happen in mainnet.

**Pull request merged**